src/types/basic/uint.ts

```diff
@@ -58,7 +55,10 @@ export class NumberUintType extends UintType<number> {
     return Number(output);
   }
   fromJson(data: Json): number {
-    const n = Number(data);
+    let n = Number(data);
+    if(n > Number.MAX_SAFE_INTEGER) {
+      n = Infinity;
```

Comment:
I think we need to be more restrictive here.
```typescript
const bigN = BigInt(data);
if (bigN === (BigInt(2) ** BigInt(this.byteLength)) - BigInt(1)) {
  n = Infinity;
} else {
  throw new Error("Uint value unsafe");
}
```
something like that, only set to Infinity if value is 0xfffffffff....